src/new-client/src/OpenFin/DetachableTile/tornOutTiles.ts

```diff
@@ -10,17 +10,36 @@ import {
 } from "rxjs/operators"
 import { CurrencyPair, currencyPairs$ } from "@/services/currencyPairs"
 import { closeWindow, openWindow } from "../utils/window"
+import { Offset } from "@/utils"
 
 type TearOutEntry = [string, boolean]
 
 const [tearOutEntry$, tearOut] = createSignal<TearOutEntry>()
 
-export const onTearOut = (symbol: string) => {
+const calculateOpeningWindowCoords = async (coords: Offset) => {
+  if (fin) {
+    const view = await fin.View.getCurrent()
+    const window = await view.getCurrentWindow()
+    const viewBounds = await view.getBounds()
+    const windowBounds = await window.getBounds()
```

Comment:
A common problem when using `async`/`await` is that we can end up making sequential requests that instead could be running in parallel. I think that this code is better in that regard:

```suggestion
    const view = await fin.View.getCurrent()

    const [viewBounds, windowBounds] = await Promise.all([
      view.getBounds(),
      view.getCurrentWindow().then(window => window.getBounds()),
    ])
```